clients/src/main/java/org/apache/kafka/common/requests/ListConfigResourcesRequest.java

```diff
@@ -36,6 +38,15 @@ public Builder(ListConfigResourcesRequestData data) {
 
         @Override
         public ListConfigResourcesRequest build(short version) {
+            if (version == 0) {
+                // The v0 only supports CLIENT_METRICS resource type.
+                // Empty resource types means all supported resource types. In v0, it means CLIENT_METRICS, so it's acceptable.
+                Set<Byte> resourceTypes = new HashSet<>(data.resourceTypes());
+                if ((resourceTypes.size() == 1 && !resourceTypes.contains(ConfigResource.Type.CLIENT_METRICS.id())) ||
```

Comment:
I think it's only valid if `resourceTypes` is non-empty (because empty means all types) and if it only contains `CLIENT_METRICS.id()`. wdyt?

Reply:
Good suggestion. Updated it. Thanks.